components/org.wso2.carbon.identity.api.server.organization.user.invitation.management/org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/invitation/management/v1/core/GuestApiServiceCore.java

```diff
@@ -0,0 +1,253 @@
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_INVALID_INVITATION_ID;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_USER_NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_IMPLEMENTED;
+
+/**
+ * Service class for the Shared User Invitation Management APIs.
+ */
+public class GuestApiServiceCore {
+
+    private static List<RoleAssignmentResponse> buildRoleAssignmentResponse(Invitation invitationRecord) {
+
+        List<RoleAssignmentResponse> roleAssignmentResponses = new ArrayList<>();
+        for (RoleAssignments roleAssignment : invitationRecord.getRoleAssignments()) {
+            if (roleAssignment.getRoles() != null) {
+                RoleAssignmentResponse roleAssignmentResponse = new RoleAssignmentResponse();
+                roleAssignmentResponse.setApplicationId(roleAssignment.getApplicationId());
+                roleAssignmentResponse.setRoles(Arrays.asList(roleAssignment.getRoles()));
+                roleAssignmentResponses.add(roleAssignmentResponse);
+            }
+        }
+        return roleAssignmentResponses;
+    }
+
+    /**
+     * Creates the invitation to the shared user.
+     *
+     * @param invitationRequestBody Contains the details of the invitation.
+     * @return The details of the created invitation.
+     */
+    public InvitationSuccessResponse createInvitation(InvitationRequestBody invitationRequestBody) {
+
+        InvitationCoreServiceImpl invitationCoreService = new InvitationCoreServiceImpl();
+        Invitation invitation = new Invitation();
+        invitation.setUsername(invitationRequestBody.getUsername());
+        invitation.setUserDomain(invitationRequestBody.getUserDomain());
+        invitation.setUserRedirectUrl(invitationRequestBody.getUserRedirectUrl());
+        if (invitationRequestBody.getRoleAssignments() != null) {
+            List<RoleAssignments> roleAssignments = new ArrayList<>();
+            for (RoleAssignmentRequestBody roleAssignmentRequestBody : invitationRequestBody.getRoleAssignments()) {
+                if (roleAssignmentRequestBody.getRoles() != null) {
+                    RoleAssignments roleAssignment = new RoleAssignments();
+                    roleAssignment.setApplicationId(roleAssignmentRequestBody.getApplicationId());
+                    roleAssignment.setRoles(roleAssignmentRequestBody.getRoles().toArray(new String[0]));
+                    roleAssignments.add(roleAssignment);
+                }
+            }
+            invitation.setRoleAssignments(roleAssignments.toArray(new RoleAssignments[0]));
```

Comment:
what about extracting this into a private method? then we use that in the previous method as well

Reply:
Unable to extract since the previous method is about the response creation and this one is for request object creation.